### ISSUE TITLE
Copying/moving folders

### ISSUE THREAD

**Issue by itsWindows11** [Arlodotexe/OwlCore.Storage#60]
There doesn't seem to be a way to copy/move a directory through the abstraction, only files in an `IModifiableFolder`. This is a necessary feature to complete the basic file management capabilities for the abstraction, and to have it integrated in Files.

Most (if not all) storage implementations we've implemented or are planning to implement already support recursively moving a folder to another folder.

**Comment by Arlodotexe** [comment]
Correct, this is by design due to some pitfalls we recognized here. 

We can't do recursive operations safely from within the abstraction. The abstraction provides an API surface of "storage primitives" such as create, enumerate, delete, read/write. But handling results and catching errors when calling these must happen on a layer just above these storage primitives, usually the library consumer or inbox tooling we've built on them (such as our various extension methods/interfaces). 

We can create optional interfaces for this, but it's a much more difficult task to write a fallback that works for _all_ possible implementations because recursion is involved. The solution to this would be complicated and likely need more specialization than we can account for at this layer.
